docs/REPRODUCIBILITY.md

```diff
@@ -0,0 +1,267 @@
+The detailed description of the yaml format can be found in [the linuxkit docs](https://github.com/linuxkit/linuxkit/blob/master/docs/yaml.md).
+
+Each container image used as an input is provided with a precise and unchanging tag.
+
+There are two image formats in the yaml file: fixed and templated.
+
+### Fixed Images
+
+Fixed tags are those that look like normal container images with tags or hashes. For example:
+
+* `linuxkit/init:8f1e6a0747acbbb4d7e24dc98f97faa8d1c6cec7`
+* `linuxkit/getty:v0.5`
+
+While tags on OCI registries, like [Docker Hub](https://hub.docker.com), are mutable, EVE only uses tags that are never mutated. This is true for both tags on `lfedge/eve-*` images, and those that come from upstream or other sources, like `linuxkit/*`.
+
+The semver tags, like `v0.5`, are never changed, nor are the hash-based tags.
+
+The hash-tags are, in all cases, the hash of the git-commited source tree of the package. Using the above example, the source to the `linuxkit/init` image
+is in this directory [https://github.com/linuxkit/linuxkit/tree/master/pkg/init](https://github.com/linuxkit/linuxkit/tree/master/pkg/init). When building the
+container image, it is tagged with the git hashed content of the directory tree. Thus, the result of that hash at the time the image was built
+was `8f1e6a0747acbbb4d7e24dc98f97faa8d1c6cec7`.
+
+It is possible to lock them down further, and make mutating tags impossible to slip through, by adding image index hashes to the yaml file, e.g.:
+
+* `linuxkit/init:8f1e6a0747acbbb4d7e24dc98f97faa8d1c6cec7@sha256:5c503aa479ae89ff7e3ea5e9d6c427e58ec78b7216ba1ad3cb2e4da2720913f0`
+* `linuxkit/getty:v0.5@sha256:6eba389196328d02ad01aea691a60931a42b6c81978a33f87e770df902869b9c`
+
+### Templated Images
+
+The other image names are templated. These are all captialized, and end in `_TAG`. For example:
+
+* `RNGD_TAG`
+* `PILLAR_TAG`
+
+These are not consumable by linuxkit; instead, the `Makefile` target that builds an EVE image, `make eve`, calls lower-level targets that parse
+this templated yaml file [rootfs.yml.in](../images/rootfs.yml.in), replacing all templates that end in `_TAG` with the appropriate image tags.
+
+These tags are calculated from the relevant directory. For example, `RNGD_TAG` is calculated based on the image that would be generated
+from the directory [pkg/rngd](../pkg/rngd/), and `PILLAR_TAG` based on the directory [pkg/pillar](../pkg/pillar/).
+
+The actual image name and tag are calculated identically to the process for the fixed tags, i.e. the git hash of the directory tree.
+The calculation is provided by running `linuxkit pkg show-tag dir/`. For example:
+
+```sh
+$ linuxkit pkg show-tag pkg/pillar 
+lfedge/eve-pillar:b20146debd08bbfffb23f53fc9f5c3c4af377f67
+```
+
+With the template complete, every OCI container image used to compose an EVE operating system is in the generated `rootfs.yml`, each of which
```

Comment:
Not clear on how we get back to this generated roofs.yml in future to find the exact hashes used in the templated images ?

Reply:
I will update the doc to show that this is the purpose of it.

Reply:
Take a look at the introduction again.